src/api/sizeme-api.js

```diff
@@ -467,10 +467,15 @@ function saveProfile () {
 
 function selectSize (size, auto) {
     return (dispatch, getState) => {
-        if (auto) {
-            SizeSelector.setSelectedSize(size);
+        const firstMatch = !uiOptions.firstRecommendation && getState().selectedSize.firstMatch;
+        if (!firstMatch) {
+            if (auto) {
+                SizeSelector.setSelectedSize(size);
+            }
+            dispatch(actions.selectSize({size, auto}));
+        } else {
+            dispatch(actions.selectSize({size: SizeSelector.getSelectedSize(), auto: false}));
         }
-        dispatch(actions.selectSize({ size, auto }));
```

Comment:
If the firstRecommendation-option is false and this is the first matching since page load, don't try to set the value of the size-selector, instead just dispatch the selectSize-action with the actual selected size. Auto is set to false at this point, since we don't know yet if the size is the recommended size (property `auto` determines if we show the recommendation splash in the UI)